### ISSUE TITLE
Scylla server start hangs

### ISSUE THREAD

**Issue by yangly0815** [scylladb/scylla#8670]
When I deploy scylla and start it, it hangs:
```
[root@test-index-58 ~]# systemctl status scylla-server
● scylla-server.service - Scylla Server
   Loaded: loaded (/usr/lib/systemd/system/scylla-server.service; disabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/scylla-server.service.d
           └─dependencies.conf, mounts.conf
   Active: activating (start) since Tue 2021-05-18 15:12:15 CST; 45min ago
  Process: 27615 ExecStopPost=/opt/scylladb/scripts/scylla_stop (code=exited, status=0/SUCCESS)
  Process: 27628 ExecStartPre=/opt/scylladb/scripts/scylla_prepare (code=exited, status=0/SUCCESS)
 Main PID: 27636 (scylla)
   Status: "initializing batchlog manager"
   CGroup: /scylla.slice/scylla-server.slice/scylla-server.service
           └─27636 /usr/bin/scylla --developer-mode 1 --alternator-address 192.168.75.131 --alternator-port 8000 --experimental-features lwt --enable-keyspace-column-family-metrics 0 --log-to-syslog 1 --l...

May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] database - Keyspace system: Reading CF schema_usertypes id=3aa75225-4f82-350b-8d5c-430fa221fa0a version=3c53e6bb-56d5-3618-a0b2-4ff5f18127d7
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] database - Keyspace system: Reading CF schema_functions id=d1b675fe-2b50-3ca4-8e49-c0f81989dcad version=a50959dd-2be0-3e4e-8bb5-cf40796e256a
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] init - starting gossip
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] init - seeds={192.168.75.129, 192.168.75.128, 192.168.75.127, 192.168.75.126, 192.168.75.125, 192.168.75.130}, listen_address=192.168.75.1...192.168.75.131
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] init - starting storage proxy
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] init - starting migration manager
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] init - starting query processor
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] init - initializing batchlog manager
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] legacy_schema_migrator - Moving 0 keyspaces from legacy schema tables to the new schema keyspace (system_schema)
May 18 15:12:16 test-index-58 scylla[27636]:  [shard 0] legacy_schema_migrator - Dropping legacy schema tables
Hint: Some lines were ellipsized, use -l to show in full.

```

**Comment by slivne** [comment]
@yangly0815 
- can you please provide the scylla version
- can you please generate a coredump from the stuck node and upload it 
to generate a coredump at this state you can send a signal ```kill -6  <scylla-pid>```

to upload https://docs.scylladb.com/troubleshooting/report_scylla_problem/

**Comment by yangly0815** [comment]
The scylla version as follows and the coredump was also uploaded.
```
[root@test-index-58 ~]# scylla --version
4.4.1-0.20210413.00da6b5e9

```

**Comment by slivne** [comment]
@elcallio can you please have an initial look at this - on why we are stuck

**Comment by elcallio** [comment]
Uhm, I can't locate a relocatable package for this. Core says it is build-id f1b7ffd311c0488984c05a1b3c0d658551419367. Neither enterprise, master, 4.4 or 4.5 seem to have such as build?

**Comment by bhalevy** [comment]
> Uhm, I can't locate a relocatable package for this. Core says it is build-id f1b7ffd311c0488984c05a1b3c0d658551419367. Neither enterprise, master, 4.4 or 4.5 seem to have such as build?

Right, so is the coredump's build-id (upload uuid `4c4ae935-02bc-46bc-b2ec-4b1150095048`)
```
$ eu-unstrip -n --core core.scylla.996.99b0a1f0142d4864a2e66888d0e69c78.7457.1621389640000000 | awk '/scylla$/ { s=$2; sub(/@.*$/, "", s); print s; exit(0); }'
f1b7ffd311c0488984c05a1b3c0d658551419367
```

Based on https://github.com/scylladb/scylla/issues/8670#issuecomment-843698308 the version is supposed to be `4.4.1-0.20210413.00da6b5e9`

But I can't find the respective relocatable packages either.

@hagitsegev @yaronkaikov were some relocatable packages removed from `http://downloads.scylladb.com.s3.amazonaws.com/unstable/scylla/branch-4.4/relocatable`?

**Comment by yaronkaikov** [comment]
@bhalevy http://downloads.scylladb.com/unstable/scylla/branch-4.4/relocatable/2021-04-10T00%3A28%3A06Z/

**Comment by bhalevy** [comment]
> @bhalevy http://downloads.scylladb.com/unstable/scylla/branch-4.4/relocatable/2021-04-10T00%3A28%3A06Z/

@yaronkaikov it has neither the right version string nor build-id.
Having the same scylla.git-sha doesn't help with debugging the core dump.
We must match the exact relocatable package with the coredump, based on matching build-id's.

**Comment by yaronkaikov** [comment]
@bhalevy I looked for the version you mention above. Are we sure it's the version? 
Anyway let me try to understand what is going on

**Comment by yangly0815** [comment]
> Uhm, I can't locate a relocatable package for this. Core says it is build-id f1b7ffd311c0488984c05a1b3c0d658551419367. Neither enterprise, master, 4.4 or 4.5 seem to have such as build?

The package was build from branch 4.4.1 by myself.

**Comment by elcallio** [comment]
Then I can't really make much of the core file. Or be very sure it is not a build error. I would suggest reproducing the problem using an official build.

**Comment by yangly0815** [comment]
> Then I can't really make much of the core file. Or be very sure it is not a build error. I would suggest reproducing the problem using an official bu

**Comment by yangly0815** [comment]
> Then I can't really make much of the core file. Or be very sure it is not a build error. I would suggest reproducing the problem using an official build.

It may not a build error, because it was not happend for every installing. And after the promblem arised, i uninstall it and reinstall, I can start the scylla server normally.

**Comment by slivne** [comment]
@yangly0815 can you elaborate why you build locally and not use our builds ?

**Comment by yangly0815** [comment]
I'm learning scylla and build it from branch-4.4. But I'm sure that I didn't modify the code.

**Comment by yangly0815** [comment]
> @yangly0815 can you elaborate why you build locally and not use our builds ?
> 
> All, we (at Scylla) work hard to test our distributed artifacts - please use them

I'm learning scylla and build it from branch-4.4. But I'm sure that I didn't modify the code.

**Comment by elcallio** [comment]
And yet I cannot debug a core file using your local build. For one, because I do not have it. You can provide all the relocatable data and debug info normally produced by a verified build, but I am unsure if we have procedures for publishing these here. Seems reproducing the issue using a provided rpm build is easier. 
Also, not modifying the code and guaranteeing that the build was proper are two different things.
